src/components/Common/EmojiPickerIconButton.tsx

```diff
@@ -0,0 +1,74 @@
+import { Box, IconButton } from "@mui/material";
+import EmojiEmotionsOutlinedIcon from "@mui/icons-material/EmojiEmotionsOutlined";
+import { useState, useEffect, useRef } from "react";
+import EmojiPicker, { EmojiClickData, EmojiStyle } from "emoji-picker-react";
+
+type EmojiPickerIconButtonProps = {
+  onEmojiClick: (emoji: EmojiClickData) => void;
+  pickerHeight?: number;
+  pickerPosition?: "top" | "bottom";
+  pickerWidth?: number;
+};
+
+const styles = {
+  emojiContainer: { position: "absolute", zIndex: 1 },
+  emojiPicker: {
+    "--epr-emoji-size": "25px",
+  },
+};
+
+const EmojiPickerIconButton = ({
+  onEmojiClick,
+  pickerHeight = 450,
+  pickerPosition = "bottom",
+  pickerWidth = 350,
+}: EmojiPickerIconButtonProps) => {
+  const emojiContainerRef = useRef<HTMLDivElement>(null);
+  const emojiIconButtonRef = useRef<HTMLButtonElement>(null);
+  const twitterEmojiStyle = EmojiStyle.TWITTER;
+  const [showEmojiPicker, setShowEmojiPicker] = useState(false);
+  const top = pickerPosition === "bottom" ? 150 : -460;
+
+  const handleClick = (event: MouseEvent) => {
+    if (
+      emojiContainerRef.current &&
```

Comment:
using the two refs is good! didnt know how to implement this so this was the right idea